src/badge/styled-components.js

```diff
@@ -0,0 +1,151 @@
+        backgroundColor: $theme.colors.backgroundInversePrimary,
+      },
+      [COLOR.positive]: {
+        color: $theme.colors.contentInversePrimary,
+        backgroundColor: $theme.colors.backgroundPositive,
+      },
+      [COLOR.negative]: {
+        color: $theme.colors.contentInversePrimary,
+        backgroundColor: $theme.colors.backgroundNegative,
+      },
+      [COLOR.warning]: {
+        color: $theme.colors.primaryA,
+        backgroundColor: $theme.colors.backgroundWarning,
+      },
+    },
+    [HIERARCHY.secondary]: {
+      [COLOR.accent]: {
+        color: $theme.colors.contentAccent,
+        backgroundColor: $theme.colors.backgroundLightAccent,
+      },
+      [COLOR.primary]: {
+        color: $theme.colors.primaryA,
+        backgroundColor: $theme.colors.backgroundSecondary,
+      },
+      [COLOR.positive]: {
+        color: $theme.colors.contentPositive,
+        backgroundColor: $theme.colors.backgroundLightPositive,
+      },
+      [COLOR.negative]: {
+        color: $theme.colors.contentNegative,
+        backgroundColor: $theme.colors.backgroundLightNegative,
+      },
+      [COLOR.warning]: {
+        color: $theme.colors.contentWarning,
+        backgroundColor: $theme.colors.backgroundLightWarning,
+      },
+    },
+  };
+
+  return COLOR_STYLES[$hierarchy][$color];
+}
+
+export const StyledRoot = styled<{}>('div', () => {
+  return {
+    position: 'relative',
+    display: 'inline-block',
+  };
+});
+
+export const StyledInlineBadge = styled<{
+  $shape?: ShapeT,
+  $color?: ColorT,
+  $hierarchy?: HierarchyT,
+  $hidden?: boolean,
+}>(
+  'div',
+  ({
+    $theme,
+    $shape = SHAPE.rectangle,
+    $color = COLOR.accent,
+    $hierarchy = HIERARCHY.primary,
+    $hidden,
+  }) => {
+    return {
+      visibility: $hidden ? 'hidden' : 'inherit',
+      boxSizing: 'border-box',
+      height: $theme.sizing.scale700,
+      borderRadius:
+        $shape === SHAPE.rectangle ? $theme.borders.radius200 : $theme.borders.radius500,
+      paddingRight: $shape === SHAPE.rectangle ? $theme.sizing.scale100 : $theme.sizing.scale300,
+      paddingLeft: $shape === SHAPE.rectangle ? $theme.sizing.scale100 : $theme.sizing.scale300,
+      display: 'inline-flex',
+      alignItems: 'center',
+      ...getColorStyles({ $theme, $hierarchy, $color }),
+      ...($hierarchy === HIERARCHY.primary
+        ? $theme.typography.LabelSmall
+        : $theme.typography.ParagraphSmall),
+    };
+  }
+);
+
+export const StyledPositioner = styled<{
+  $role: RoleT,
+  $placement: PlacementT,
+  $horizontalOffset: ?OffsetT,
+  $verticalOffset: ?OffsetT,
+}>('div', ({ $theme, $role, $placement, $horizontalOffset, $verticalOffset }) => {
+  let positionStyle = POSITION_STYLES[$placement][$role];
+
+  if ($verticalOffset) {
+    if (
+      $placement === PLACEMENT.topLeft ||
+      $placement === PLACEMENT.top ||
+      $placement === PLACEMENT.topRight
+    ) {
+      positionStyle = { ...positionStyle, top: $verticalOffset };
```

Comment:
nit, but no need to spread when only assigning a value to a property is required. similar could be done for the rest of these cases

```suggestion
      positionStyle.top = $verticalOffset;
```